frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx

```diff
@@ -439,146 +439,132 @@ export const ProvidedAPIPage = connectToModel((props: ProvidedAPIPageProps) => {
-              value={blockValue(currentStatus, status)}
+              descriptor={statusDescriptor}
+              value={_.get(obj.status, statusDescriptor.path)}
+              obj={obj}
+              model={kindObj}
             />
           </div>
-        )}
-
-        {specDescriptors.map((specDescriptor: Descriptor) => (
-          <div key={specDescriptor.path} className="col-sm-6">
-            <SpecDescriptor
-              namespace={metadata.namespace}
-              obj={props.obj}
-              model={props.kindObj}
-              value={blockValue(specDescriptor, spec)}
-              descriptor={specDescriptor}
-              onHandleError={handleError}
-            />
-          </div>
-        ))}
-
-        {statusDescriptors
-          .filter(function(descriptor) {
-            return !isMainDescriptor(descriptor) && descriptor.displayName !== 'Status';
-          })
-          .map((statusDescriptor: Descriptor) => {
-            const statusValue = blockValue(statusDescriptor, status);
-            return (
-              <div className="col-sm-6" key={statusDescriptor.path}>
-                <StatusDescriptor
-                  namespace={metadata.namespace}
-                  obj={props.obj}
-                  model={props.kindObj}
-                  descriptor={statusDescriptor}
-                  value={statusValue}
-                />
-              </div>
-            );
-          })}
-      </div>
+        );
+      })}
     </div>
-  );
+  ) : null;
+
+export const OperandDetails = connectToModel(
+  ({ clusterServiceVersion, kindObj, obj }: OperandDetailsProps) => {
+    const { kind, status } = obj;
+    const [errorMessage, setErrorMessage] = React.useState(null);
+    const handleError = (message: string) => setErrorMessage(message);
+
+    // Find the matching CRD spec for the kind of this resource in the CSV.
+    const { displayName, specDescriptors = [], statusDescriptors = [] } =
+      _.find(
+        [
+          ...(clusterServiceVersion?.spec?.customresourcedefinitions?.owned ?? []),
+          ...(clusterServiceVersion?.spec?.customresourcedefinitions?.required ?? []),
+        ],
+        (def) => def.name.split('.')[0] === kindObj.plural,
+      ) ?? {};
+
+    const { podStatusDescriptors = [], otherStatusDescriptors = [] } = _.reduce<any, any>(
+      statusDescriptors,
+      (descriptorAccumulator, descriptor) => {
+        const { 'x-descriptors': capabilities = [] } = descriptor;
+        // exclude Conditions since they are included in their own section
+        if (descriptor.path === 'conditions') {
+          return descriptorAccumulator;
+        }
 
-  return (
-    <div className="co-operand-details co-m-pane">
-      {_.isEmpty(primaryDescriptors) ? (
+        // pod status descriptors are a special case since they are always rendered at the top, above
+        // the rest of the details
+        if (capabilities.includes(StatusCapability.podStatuses)) {
```

Comment:
If you use optional chaining here, can we avoid the `= []` defaulting above? Optional chaining handles `null`.

```suggestion
        if (capabilities?.includes(StatusCapability.podStatuses)) {
```